contracts/JBTiered721Delegate.sol

```diff
@@ -1,776 +1,632 @@
+    /// @param _interfaceId The ID of the interface to check for adherence to.
+    function supportsInterface(bytes4 _interfaceId) public view override returns (bool) {
+        return _interfaceId == type(IJBTiered721Delegate).interfaceId || super.supportsInterface(_interfaceId);
     }
-    if (_encodedIPFSUriTierId != 0 && _encodedIPFSUri != bytes32(0)) {
-      // Store the new value.
-      store.recordSetEncodedIPFSUriOf(_encodedIPFSUriTierId, _encodedIPFSUri);
 
-      emit SetEncodedIPFSUri(_encodedIPFSUriTierId, _encodedIPFSUri, msg.sender);
+    //*********************************************************************//
+    // -------------------------- constructor ---------------------------- //
+    //*********************************************************************//
+
+    /// @param _projects A contract which mints ERC-721's that represent project ownership and transfers.
+    /// @param _operatorStore A contract storing operator assignments.
+    constructor(IJBProjects _projects, IJBOperatorStore _operatorStore) JBOwnable(_projects, _operatorStore) {
+        codeOrigin = address(this);
     }
-  }
 
-  //*********************************************************************//
-  // ----------------------- public transactions ----------------------- //
-  //*********************************************************************//
+    /// @notice Initializes a cloned copy of this contract.
+    /// @param _projectId The ID of the project this contract's functionality applies to.
+    /// @param _directory The directory of terminals and controllers for projects.
+    /// @param _name The name of the collection distributed through this contract.
+    /// @param _symbol The symbol that the collection should be represented by.
+    /// @param _fundingCycleStore A contract storing all funding cycle configurations.
+    /// @param _baseUri A URI to use as a base for full token URIs.
+    /// @param _tokenUriResolver A contract responsible for resolving the token URI for each token ID.
+    /// @param _contractUri A URI where contract metadata can be found.
+    /// @param _pricing The tier pricing according to which token distribution will be made. Must be passed in order of contribution floor, with implied increasing value.
+    /// @param _store A contract that stores the NFT's data.
+    /// @param _flags A set of flags that help define how this contract works.
+    function initialize(
+        uint256 _projectId,
+        IJBDirectory _directory,
+        string memory _name,
+        string memory _symbol,
+        IJBFundingCycleStore _fundingCycleStore,
+        string memory _baseUri,
+        IJB721TokenUriResolver _tokenUriResolver,
+        string memory _contractUri,
+        JB721PricingParams memory _pricing,
+        IJBTiered721DelegateStore _store,
+        JBTiered721Flags memory _flags
+    ) public override {
+        // Make the original un-initializable.
+        if (address(this) == codeOrigin) revert();
```

Comment:
Is this needed? There is no selfdestruct, having an initialised original doesn't change much

Reply:
deleted